packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -271,6 +273,9 @@ export default class DatagridController {
             this.displayedRows = DatagridController.createEmptyRows(this.paging.getCurrentPageSize());
         }
 
+        this.selectedRows = this.selectedRows.map(() => false);
```

Comment:
Can't this contain the actual rows instead of an array of booleans ? Or something to identify the rows ?

```
{
	id: rowId,
	isSelected: false
}
```
If we keep the array of booleans system, then it should be names something like `rowSelectionStatus`

Reply:
since every checkbox needs to have an ng-model i still have to created N booleans for binding ; that's why it made sense to use an array of booleans (i reuse the array)

Reply:
We could react to the checkbox's `on-change` event and add the corresponding row to the array of rows.

Reply:
i think your suggestion would make the code "faster" : no reduce in getSelectedRows since you access array directly ; downside is that you need an additionnal array to store rows which is less clean (you still have to give an ngModel to checkbox so you also need an array of booleans).
did i understand you correcly ?